### PR TITLE
Enable accessing InTransit/Receiving boxes for members of target base

### PR DESCRIPTION
**Pull request by pylipp**
New behavior: If I access an InTransit or Receiving box, and I'm member of either source or target base of the corresponding shipment, 
- no authz error is thrown, and
- `Box.product` and `Box.location` return null if the box is still registered in the opposite base. I have to use `ShipmentDetail.source/targetProduct/Location` instead

### CODE REVIEW

**Comment by pylipp** [comment]
[2.0 Enable accessing InTransit/Receiving boxes for members of target base](https://trello.com/c/5ePQDQ1g/1237-20-enable-accessing-intransit-receiving-boxes-for-members-of-target-base)

**Review by HaGuesto — APPROVED** [review]
works as expected.

One caveat though: By chance I did the functional test with a box that was marked as not delivered. It is an edge case. But in general, no clue what happens if you find a box that was previously not delivered.

File: back/boxtribute_server/graph_ql/loaders.py [hunk 1]
```diff
@@ -42,10 +42,7 @@ async def batch_load_fn(self, keys):
 class LocationLoader(DataLoader):
     async def batch_load_fn(self, keys):
         locations = {
-            loc.id: loc
-            for loc in Location.select().where(
-                Location.id << keys, authorized_bases_filter(Location)
```
[the remark this answers is not in the document]
Comment (pylipp) [reply back/boxtribute_server/graph_ql/loaders.py:47]: @HaGuesto that's correct. When a user is not authzed for a location/product, null will be returned from the `Box.location` / `Box.product` fields.

**Comment by pylipp** [comment]
> One caveat though: By chance I did the functional test with a box that was marked as not delivered. It is an edge case. But in general, no clue what happens if you find a box that was previously not delivered.

@HaGuesto good point; right now a `Lost` box can only be accessed by the people who are authorized for the box's base. If a box is marked as lost in the course of a shipment, that will be the source base. So the target base won't be able to access the box. However in #839 we work around this by introducing the `Box.NotDelivered` state.
